pkg/cmd/agent-task/list/list.go

```diff
@@ -44,6 +44,9 @@ func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Comman
 			if f != nil {
 				opts.BaseRepo = f.BaseRepo
 			}
+			if opts.Limit < 1 {
+				return cmdutil.FlagErrorf("invalid limit: %v", opts.Limit)
+			}
```

Comment:
Also, the limit should not be negative. We can either use `cmd.Flags().UintVarP` or just check for negative values here. I'd personally go with the former, which also requires us to either update the underlying `limit` parameter type in `CapiClient` methods, or just do a type conversion (i.e. `uint(limit)`) at the calling time.

I'm fine either way, so it's up to you.

Reply:
My bad, my brain must have stopped at this point. :smile: